Model/BoldQuoteRepository.php

```diff
@@ -0,0 +1,37 @@
+<?php
+//phpcs:disable Magento2.Annotation.MethodArguments.NoCommentBlock
+//phpcs:disable Magento2.Annotation.MethodArguments.ParamMissing
+//phpcs:disable Magento2.Annotation.MethodAnnotationStructure.MethodAnnotation
+
+declare(strict_types = 1);
+
+namespace Bold\Checkout\Model;
+
+use Bold\Checkout\Api\BoldQuoteRepositoryInterface;
+use Bold\Checkout\Api\Data\BoldQuoteInterface;
+use Bold\Checkout\Model\ResourceModel\Quote\QuoteExtensionData as QuoteExtensionDataResource;
+use Bold\Checkout\Model\Quote\QuoteExtensionDataFactory;
+use Magento\Framework\Exception\NoSuchEntityException;
+
+class BoldQuoteRepository implements BoldQuoteRepositoryInterface
+{
+    public function __construct(
+        private readonly QuoteExtensionDataResource $boldQuoteResource,
+        private readonly QuoteExtensionDataFactory $quoteExtensionDataFactory,
```

Comment:
Backwards compatibility: Constructor property promotion and the `readonly` modifier are PHP 8 features that will break compatibility with PHP 7 (we support back to 7.2). 

**Hint**: There is a [Rector](https://getrector.com) configuration file in the root that will check for PHP compatibility issues. Note that syntax formatting will need to be adjusted to match PSR-12 or PER-CS after running this tool.